packages/fxa-shared/metrics/amplitude.js

```diff
@@ -335,3 +336,20 @@ function mapNewsletterState (eventCategory, data) {
     return { newsletter_state };
   }
 }
+
+function mapNewsletters (data) {
+  const {newsletters} = data;
+  const currentNewsletters = {};
+
+  if (newsletters) {
+    for (const key in newsletters) {
+      const newsletter = newsletters[key];
+      if (newsletter === true || newsletter === false) {
+        currentNewsletters[toSnakeCase(key)] = newsletters[key] ? 'subscribed' : 'unsubscribed';
+      }
+    }
+    return {
+      newsletters: currentNewsletters
+    };
```

Comment:
I'm not really a fan of procedural loops so wonder if this would read clearer as a `reduce`. Something like (untested):

```js
return {
  newsletters: Object.values(newsletters).reduce((result, { key, isSubscribed }) => {
    if (isSubscribed === true || isSubscribed === false) {
      result[toSnakeCase(key)] = isSubscribed ? 'subscribed' : 'unsubscribed';
      return result;
  }, {})
};
```

Reply:
With https://github.com/mozilla/fxa/pull/1177#issuecomment-494439163, this got a little easier since we now pass a list of newsletters.